src/material/list/_list-theme.scss

```diff
@@ -1,25 +1,28 @@
 @use 'sass:map';
-@use '@material/list/evolution-mixins' as mdc-list;
+@use '@material/list/evolution-mixins';
+@use '@material/list/list-theme' as mdc-list-theme;
 
 @use '../core/theming/theming';
+@use '../core/tokens/m2/mdc/list' as m2-mdc-list;
 @use '../core/typography/typography';
 @use '../core/typography/typography-utils';
 @use '../core/mdc-helpers/mdc-helpers';
-@use './interactive-list-theme';
 @use './list-option-theme';
 
 @mixin color($config-or-theme) {
   $config: theming.get-color-config($config-or-theme);
   $primary: theming.get-color-from-palette(map.get($config, primary));
   $accent: theming.get-color-from-palette(map.get($config, accent));
   $warn: theming.get-color-from-palette(map.get($config, warn));
+  $mdc-list-color-tokens: m2-mdc-list.get-color-tokens($config);
 
-  @include mdc-helpers.using-mdc-theme($config) {
-    // MDC's state styles are tied in with their ripple. Since we don't use the MDC
-    // ripple, we need to add the hover, focus and selected states manually.
-    @include interactive-list-theme.private-interactive-list-item-state-colors($config);
-    @include mdc-list.without-ripple($query: mdc-helpers.$mdc-theme-styles-query);
+  // Add values for MDC list tokens.
+  .mat-mdc-list-base {
+    @include mdc-list-theme.theme($mdc-list-color-tokens);
```

Comment:
Are we sure that this is generating variables? I'm seeing a bunch of declarations like this in the theme:

```
.mdc-list-item__primary-text {
    color: var(--mdc-theme-text-primary-on-background, rgba(0, 0, 0, 0.87))
}
```

Reply:
I'll request a demo for you to poke at, but yes - you may see code like that in the demo. It comes from other components that are still using the old MDC mixins. I confirmed that the theme mixin really is outputting the correct CSS variable defs, you can find them if you search for one of the token names in the CSS

Reply:
I see. A couple of follow-ups:
1. Should we scope those other components under a different selector? I can see this being a problem where the styles bleed in from other components.
2. I can see that the variables are being generated under selectors like this. Wasn't the idea to output them under the `:root`?
![hprmdfe](https://user-images.githubusercontent.com/4450522/217347616-2dbbcfb7-54eb-4038-a0c0-ddd2396a0863.png)

Reply:
1. We could try, but it will probably be difficult to land because its will bump the specificity of everything and break people's overrides.
2. I think that example you posted is maybe from some pre-existing checkbox code. I'm only outputting `--mdc-list-*` variables in this PR. I'm outputting them under a selector for the host element: `.mat-mdc-list-base { ... }`. We may want to move them to the root later, but this feels like the safest thing to do for now